src/pages/main/components/selected-dataset/build-flow/components/build-flow-header.tsx

```diff
@@ -0,0 +1,95 @@
+import styles from '../build-flow.module.css'
+
+import { ReactElement } from 'react'
+import { useHistory } from 'react-router'
+import { observer } from 'mobx-react-lite'
+
+import { t } from '@i18n'
+import { datasetStore } from '@store/dataset'
+import dirinfoStore from '@store/dirinfo'
+import { Routes } from '@router/routes.enum'
+import { CardTitle } from '@ui/card'
+import { Icon } from '@ui/icon'
+import { DatasetCard } from '@components/data-testid/dataset-card.cy'
+import wizardStore from './wizard/wizard.store'
+
+export const BuildFlowHeader = observer(
+  ({ goBack }: { goBack: () => void }): ReactElement => {
+    const history = useHistory()
+    const isXL = datasetStore.isXL
+
+    const handleGoMain = () => {
+      goBack()
+      history.push(Routes.Root)
+      wizardStore.actionHistory.resetHistory()
+      wizardStore.resetScenario()
+    }
+
+    const handleGoStart = () => {
+      goBack()
+      wizardStore.actionHistory.resetHistory()
+      wizardStore.resetScenario()
+    }
+
+    const { selectedDataset, datasetName } = wizardStore
+
+    const handleGoBack = () => {
+      if (wizardStore.actionHistory.historyIndex === 0) {
+        isXL ? handleGoStart() : handleGoMain()
+      } else {
+        wizardStore.actionHistory.goBackward()
+      }
+    }
```

Comment:
this function can be rewritten with using an early return pattern instead of nested conditions